haskell-style.md

```diff
@@ -41,6 +41,18 @@ done for a few reasons:
 - This gives us a chance to see just how much time we've saved by automating
   what used to be a (possibly tenuous) negotiated consensus
 
+## Whitespace
+
+- Prefer two space indentation
+
+  Exceptions include anything automated (e.g. de-dented `where`) and the leading
+  tabs required in `Makefile`s.
+
+- No trailing whitespace
+
+  No extra spaces at the end of lines, and no extra lines at the end of the file
```

Comment:
Brittany takes care of the former, but not the latter. Prettier is handling both on the JS side. There is almost certainly other language-agnostic plugins to handle it, worth finding and mentioning?

Reply:
My google-fu may be busted, but I'm not finding a definitive way of doing this in `vim`.

Reply:
Did you find a way to make restyled do this?

https://github.com/freckle/megarepo/pull/9793

Reply:
Huh, I can't seem to find a plugin that mentions that feature explicitly either. I would _guess_:

1. You're in a minority as someone who, for some reason, frequently hits enter an extra time when you are done editing the last night of a file
1. Plugins (like prettier) happen to handle this *by accident* depending on how they implement the output of the reformatted AST, but never call it out as an explicit feature, because of (1)

Reply:
Oh wait no, that is from a garbo solution I was trying in `vim`.

Reply:
> Note that this is not "no end of line", and your vim setting above is incorrect. That would create files with no trailing newline, and that could cause actual technical problems. They will cause warnings in git, for example. You can spot the difference in GitHub too, it'll say in red No newline or some such.

That was Restyled (rather Brittany, I think), fixing that mistake.